mirgecom/viscous.py

```diff
@@ -221,23 +222,46 @@ def viscous_facial_flux(discr, eos, cv_tpair, grad_cv_tpair,
     return flux_weak
 
 
-def get_viscous_timestep(discr, eos, transport_model, cfl, cv):
-    """Routine (will) return the (local) maximum stable viscous timestep.
+def get_viscous_timestep(discr, eos, cv):
+    """Routine returns the the node-local maximum stable viscous timestep.
 
-    Currently, it's a hack waiting for the geometric_factor helpers port
-    from grudge.
+    Similar to get_inviscid_timestep but there's a viscosity term.
     """
-    dim = discr.dim
-    mesh = discr.mesh
-    order = max([grp.order for grp in discr.discr_from_dd("vol").groups])
-    nelements = mesh.nelements
-    nel_1d = nelements ** (1.0 / (1.0 * dim))
-
-    # This roughly reproduces the timestep AK used in wave toy
-    dt = (1.0 - 0.25 * (dim - 1)) / (nel_1d * order ** 2)
-    return cfl * dt
-#    dt_ngf = dt_non_geometric_factor(discr.mesh)
-#    dt_gf  = dt_geometric_factor(discr.mesh)
-#    wavespeeds = compute_wavespeed(w,eos=eos)
-#    max_v = clmath.max(wavespeeds)
-#    return c*dt_ngf*dt_gf/max_v
+    from grudge.dt_utils import characteristic_lengthscales
+    from mirgecom.fluid import compute_wavespeed
+
+    length_scales = characteristic_lengthscales(cv.array_context, discr)
+
+    mu = 0
+    transport = eos.transport_model()
+    if transport:
+        mu = transport.viscosity(eos, cv)
+
+    weighted_viscosity = (mu / length_scales)
+
+    return (
+        length_scales / (compute_wavespeed(eos, cv) + weighted_viscosity)
+    )
```

Comment:
These little temporaries make it easier to read and understand, but we avoid them in this code whenever possible.
```suggestion
    return(
        length_scales / (compute_wavespeed(eos, cv) + mu / length_scales)
    )
```